src/main/java/core/basesyntax/GeometricFigureSupplier.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class GeometricFigureSupplier {
+    private Random random = new Random();
+
+    public GeometricFigure getRandomFigure() {
+        int index = random.nextInt(0, GeometricFigure.values().length);
+        return GeometricFigure.values()[index];
```

Comment:
Calling 'GeometricFigure.values()' inside a loop or a frequently called method can be inefficient because it creates a new array each time it is called. Consider storing the values in a constant array and reusing it.